clientconn.go

```diff
@@ -118,6 +118,15 @@ func WithPicker(p Picker) DialOption {
 // WithBackoffConfig configures the dialer to use the provided backoff
 // parameters after connection failures.
 func WithBackoffConfig(b *BackoffConfig) DialOption {
```

Comment:
I suppose that works. Don't we want to apply defaults anyways if a partial backoffconfig is provided?